pymc/distributions/distribution.py

```diff
@@ -542,15 +546,45 @@ def dist(
         ndim_supp = cls.ndim_supp(*dist_params)
         create_size = find_size(shape=shape, size=size, ndim_supp=ndim_supp)
         rv_out = cls.rv_op(*dist_params, size=create_size, **kwargs)
-        # This is needed for resizing from dims in `__new__`
-        rv_out.tag.ndim_supp = ndim_supp
 
         rv_out.logp = _make_nice_attr_error("rv.logp(x)", "pm.logp(rv, x)")
         rv_out.logcdf = _make_nice_attr_error("rv.logcdf(x)", "pm.logcdf(rv, x)")
         rv_out.random = _make_nice_attr_error("rv.random()", "pm.draw(rv)")
         return rv_out
 
 
+class SymbolicRandomVariable(OpFromGraph):
+    """TODO"""
+
+    ndim_supp = None
+
+    def __init__(self, *args, ndim_supp, **kwargs):
+        self.ndim_supp = ndim_supp
+        kwargs.setdefault("inline", True)
+        super().__init__(*args, **kwargs)
+
+
+@node_rewriter([SymbolicRandomVariable])
+def inline_SymbolicRandomVariable(fgraph, node):
+    """
+    This optimization expands the internal graph of a SymbolicRV when obtaining logp
+    from Aeppl, in the case it does not have a specialized logp associated with it.
+    """
+    op = node.op
+    # If `Op` is not a `MeasurableVariable`, Aeppl must derive its logp from the inner graph
+    if not isinstance(op, MeasurableVariable):
+        return clone_replace(op.inner_outputs, {u: v for u, v in zip(op.inner_inputs, node.inputs)})
+
+
+# Registered before pre-canonicalization which happens at position=-10
+logprob_rewrites_db.register(
+    "inline_SymbolicRandomVariable",
+    in2out(inline_SymbolicRandomVariable),
+    "basic",
+    position=-20,
```

Comment:
I don’t really understand what the rewrite_db arguments mean. Could you explain these to me?

Reply:
I can add kwargs to make it more readable. It's just name, rewrite, tag (used for manually including or excluding rewrites), and order in which rewrite is applied from all the rewrites in the database